codex-rs/core/tests/suite/rmcp_client.rs

```diff
@@ -145,18 +149,210 @@ async fn rmcp_tool_call_round_trip() -> anyhow::Result<()> {
         .get("echo")
         .and_then(Value::as_str)
         .expect("echo payload present");
-    assert_eq!(echo_value, "ping");
+    assert_eq!(echo_value, "ECHOING: ping");
     let env_value = map
         .get("env")
         .and_then(Value::as_str)
         .expect("env snapshot inserted");
     assert_eq!(env_value, expected_env_value);
 
-    let task_complete_event =
-        wait_for_event(&fixture.codex, |ev| matches!(ev, EventMsg::TaskComplete(_))).await;
-    eprintln!("task_complete_event: {task_complete_event:?}");
+    wait_for_event(&fixture.codex, |ev| matches!(ev, EventMsg::TaskComplete(_))).await;
 
     server.verify().await;
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
```

Comment:
Prefer single thread unless you know you need it because multi thread can hide bugs.